examples/pentree.py

```diff
@@ -0,0 +1,388 @@
+    batch pos 2  [7, 8, 9]   [10, 11, 12]
+
+    This ensures that we use the last hidden state of batch 1 to initialize
+    batch 2.
+
+    Also creates targets. In language modelling the target is to predict the
+    next word in the sequence.
+
+    Parameters
+    ----------
+    x_in : 1D numpy.array
+    batch_size : int
+    model_seq_len : int
+        number of steps the model is unrolled
+
+    Returns
+    -------
+    reordered x_in and reordered targets. Targets are shifted version of x_in.
+
+    """
+    if x_in.ndim != 1:
+        raise ValueError("Data must be 1D, was", x_in.ndim)
+
+    if x_in.shape[0] % (batch_size*model_seq_len) == 0:
+        print(" x_in.shape[0] % (batch_size*model_seq_len) == 0 -> x_in is "
+              "set to x_in = x_in[:-1]")
+        x_in = x_in[:-1]
+
+    x_resize =  \
+        (x_in.shape[0] // (batch_size*model_seq_len))*model_seq_len*batch_size
+    n_samples = x_resize // (model_seq_len)
+    n_batches = n_samples // batch_size
+
+    targets = x_in[1:x_resize+1].reshape(n_samples, model_seq_len)
+    x_out = x_in[:x_resize].reshape(n_samples, model_seq_len)
+
+    out = np.zeros(n_samples, dtype=int)
+    for i in range(n_batches):
+        val = range(i, n_batches*batch_size+i, n_batches)
+        out[i*batch_size:(i+1)*batch_size] = val
+
+    x_out = x_out[out]
+    targets = targets[out]
+
+    return x_out.astype('int32'), targets.astype('int32')
+
+
+def traindata(model_seq_len, batch_size, vocab_map, vocab_idx):
+    x = load_data(os.path.join(folder, "ptb.train.txt.gz"),
+                  vocab_map, vocab_idx)
+    return reorder(x, batch_size, model_seq_len)
+
+
+def validdata(model_seq_len, batch_size, vocab_map, vocab_idx):
+    x = load_data(os.path.join(folder, "ptb.valid.txt.gz"),
+                  vocab_map, vocab_idx)
+    return reorder(x, batch_size, model_seq_len)
+
+# vocab_map and vocab_idx are updated as side effects of load_data
+vocab_map = {}
+vocab_idx = [0]
+x_train, y_train = traindata(MODEL_SEQ_LEN, BATCH_SIZE, vocab_map, vocab_idx)
+x_valid, y_valid = validdata(MODEL_SEQ_LEN, BATCH_SIZE, vocab_map, vocab_idx)
+vocab_size = vocab_idx[0]
+
+
+print("-" * 80)
+print("Vocab size:s", vocab_size)
+print("Data shapes")
+print("Train data:", x_train.shape)
+print("Valid data:", x_valid.shape)
+print("-" * 80)
+
+# Theano symbolic vars
+sym_x = T.imatrix()
+sym_y = T.imatrix()
+
+# symbolic vars for initial recurrent initial states
+cell1_init_sym = T.matrix()
+hid1_init_sym = T.matrix()
+cell2_init_sym = T.matrix()
+hid2_init_sym = T.matrix()
+
+
+# BUILDING THE MODEL
+# Model structure:
+#
+#    embedding --> LSTM1 --> LSTM2 --> output network --> predictions
+l_inp = lasagne.layers.InputLayer((BATCH_SIZE, MODEL_SEQ_LEN))
+
+l_emb = lasagne.layers.EmbeddingLayer(
+    l_inp,
+    input_size=vocab_size,       # size of embedding = number of words
+    output_size=embedding_size,  # vector size used to represent each word
+    W=INI)
+
+if dropout_frac > 0:
```

Comment:
this shouldn't be necessary, a dropout layer is just a pass through if the dropout fraction is 0, so it doesn't add any overhead.